src/Pmi/PatientStatus/PatientStatus.php

```diff
@@ -206,6 +222,56 @@ public function hasAccess($participant)
             !$this->app->isDVType() &&
             $participant->statusReason !== 'withdrawal' &&
             $participant->statusReason !== 'test-participant' &&
-            !$this->app->isTestSite();
+            !$this->app->isTestSite() &&
+            empty($this->app->getConfig('disable_patient_status_message'));
+    }
+
+    public function loadData($participantId, $patientStatusId, $form)
```

Comment:
Instead of passing in a Form object, I would call $form->getData() from outside this method and then pass it in.  That way, this method isn't tightly coupled with the Form object.